backend/scripts/import-posts/index.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable no-console */
+
+const mongoose = require("mongoose");
+const Airtable = require("airtable");
+require("dotenv").config();
+
+const {
+  config: { mongo, airtable },
+} = require("../../config");
+require("../../lib/models/User");
+const { schema: userSchema } = require("../../lib/models/User");
+const { schema: postSchema } = require("../../lib/models/Post");
+const mapAirtableData = require("./map-airtable-data");
+
+const POSTS_AIRTABLE_NAME = "Posts";
+const SOURCED_BY_FP_USER = {
+  name: "Sourced by FightPandemics",
+};
+const DEFAULT_MAX_RECORDS = 100;
+
+/* TODO: consider library (commander, yargs)
+  For now single positional arg to control records fetched:
+    a) all records: 'npm run import-posts -- -1';
+    b) specify X >= 1: 'npm run import-posts -- X';
+    c) default: 'npm run import-posts' */
+const parseMaxRecordsArg = () => {
+  const maxRecordsArg = process.argv[2];
+  return parseInt(maxRecordsArg, 10) || DEFAULT_MAX_RECORDS;
+};
+
+const importPostsFromAirtable = async (connection, sourcedById) => {
+  const start = Date.now();
+
+  const base = new Airtable({
+    apiKey: airtable.apiKey,
+  }).base(airtable.baseId);
+
+  const Post = connection.model("Post", postSchema);
+  let newPosts = 0;
+  let existingPosts = 0;
+  let failedPosts = 0;
+
+  const selectOptions = { maxRecords: parseMaxRecordsArg() };
+  const records = await base(POSTS_AIRTABLE_NAME).select(selectOptions).all(); // ~12s to get & (locally) import ~2500 posts (~8s to get from Airtable)
+
+  // eslint-disable-next-line no-restricted-syntax
+  for await (const record of records) {
+    try {
+      const res = await Post.findOneAndUpdate(
+        { airtableId: record.id },
+        mapAirtableData(record._rawJson, {
+          id: sourcedById,
+          name: SOURCED_BY_FP_USER.name,
+        }),
+        { new: true, upsert: true, useFindAndModify: false }
+      );
+      res.updatedAt.toString() === res.createdAt.toString() // eslint-disable-line no-unused-expressions
+        ? (newPosts += 1)
+        : (existingPosts += 1);
+    } catch (err) {
+      failedPosts += 1;
+      console.error(
+        `${record.id} failed to import with fields: ${JSON.stringify(
+          record._rawJson.fields
+        )}. Error message: ${err.message}`
+      );
+    }
+  }
```

Comment:
What I anticipate is in production, since we're sending the request from AWS to MongoDB Atlas one at a time, network latency could become a significant factor for my current script (while on local docker network it isn't). 

If we're only doing this a handful of times in production I don't want to spend too much time on optimization until we need to, but I'll look into this.

Reply:
Got it working and it seems to work nicely. It cut time locally in ~half but in production it should have a significant impact.

Reply:
@joshmorel - just wanted to mention this in regards to your comment about latency between AWS and MongoDB, we will have a [VPC peering connection](https://docs.atlas.mongodb.com/security-vpc-peering/) between the two which runs on the global AWS backbone connection (meaning, the traffic stays within the AWS network, and never goes across the internet). But it's always nice to optimize regardless.